packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -473,6 +473,124 @@ describe("Presence", () => {
+						averageLatency: 50,
+						clientSessionId: secondAttendeeSessionId,
+						clientConnectionId: secondAttendeeConnectionId,
+						updateProviders: ["client4"],
+					});
+
+					firstAttendeeSignalWithUpdate = generateBasicClientJoin(clock.now - 50, {
+						averageLatency: 50,
+						clientSessionId: firstAttendeeSessionId,
+						clientConnectionId: firstAttendeeConnectionId,
+						priorClientToSessionId: {
+							...firstAttendeeSignal.content.data["system:presence"].clientToSessionId,
+							[secondAttendeeConnectionId]: {
+								rev: 0,
+								timestamp: 0,
+								value: secondAttendeeSessionId,
+							},
+						},
+					});
+				});
+
+				describe("that are joining", () => {
+					let newAttendees: ISessionClient[];
+					beforeEach(() => {
+						newAttendees = [];
+						afterCleanUp.push(
+							presence.events.on("attendeeJoined", (attendee) => {
+								assert(newAttendees.length < 3, "Only two attendees should be announced");
+								newAttendees.push(attendee);
+							}),
+						);
+					});
+
+					function verifyNewAttendees(
+						expectedConnectionId1: ClientConnectionId,
+						expectedConnectionId2: ClientConnectionId,
+					): void {
+						assert.equal(newAttendees.length, 2, "Not all attendees were announced");
+						assert.ok(newAttendees[0]);
+						assert.ok(newAttendees[1]);
+						assert.equal(
+							newAttendees[0].sessionId,
+							firstAttendeeSessionId,
+							"First attendee has wrong session id",
+						);
+						assert.equal(
+							newAttendees[0].getConnectionId(),
+							expectedConnectionId1,
+							"First attendee has wrong client connection id",
+						);
+						assert.equal(
+							newAttendees[0].getConnectionStatus(),
+							SessionClientStatus.Connected,
+							"First attendee connection status is not Connected",
+						);
+						assert.equal(
+							newAttendees[1].sessionId,
+							secondAttendeeSessionId,
+							"Second attendee has wrong session id",
+						);
+						assert.equal(
+							newAttendees[1].getConnectionId(),
+							expectedConnectionId2,
+							"Second attendee has wrong client connection id",
+						);
+						assert.equal(
+							newAttendees[1].getConnectionStatus(),
+							SessionClientStatus.Connected,
+							"Second attendee connection status is not Connected",
+						);
+					}
+
+					it("is announced via `attendeeJoined`", () => {
+						// Act - simulate join message from clients
+						presence.processSignal("", firstAttendeeSignal, false);
+						presence.processSignal("", secondAttendeeSignal, false);
+						presence.processSignal("", firstAttendeeSignalWithUpdate, false);
+
+						// Verify
+						verifyNewAttendees(firstAttendeeConnectionId, secondAttendeeConnectionId);
+					});
+
+					it("is announced via `attendeeJoined` when second joining attendee is unknown to audience", () => {
+						// Setup - remove connection from audience
+						runtime.removeMember(secondAttendeeConnectionId);
```

Comment:
Could be an improvement to use a connection id for second that is not in the audience to begin with. Then there aren't any removeMember events flowing to presence. If you use a setup helper it could have the option of second client connection id [and work for both cases].